jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/admin/commands/TriggerNodeCommand.java

```diff
@@ -62,10 +65,18 @@ public Void execute(Context context) {
             if (node == null) {
                 throw new NodeNotFoundException("Node with id " + nodeId + " not found");
             }
-            NodeInstanceContainer nodeInstanceContainerNode =
-                    ((NodeInstanceContainer) wfp.getNodeInstance(wfp.getRuleFlowProcess().getParentNode(nodeId)));
+
+            Node parentNode = wfp.getRuleFlowProcess().getParentNode(nodeId);
+            // we need to unwrap the instance otherwise we are not getting the right nodeInstance
+            Map<String, Object> parentNodeMetaData = parentNode.getMetaData();
+            Boolean oldWrapValue = (Boolean) parentNodeMetaData.get(NodeInstance.METADATA_WRAP_ASYNC_NODE);
+            parentNodeMetaData.put(NodeInstance.METADATA_WRAP_ASYNC_NODE, false);
```

Comment:
this is not thread safe operation as the node is shared across threads running in parallel. So changing it here and another thread for another process instance can change it back before the node instance is retrieved. I'd create new method in workflowprocessinstance that can to the explicit unwrap.